metagraph/modules/l0/src/main/scala/com/my/dor_metagraph/l0/Main.scala

```diff
@@ -145,7 +148,19 @@ object Main extends CurrencyL0App(
   override def dataApplication: Option[Resource[IO, BaseDataApplicationL0Service[IO]]] =
     makeL0Service.asResource.some
 
-  override def rewards(implicit sp: SecurityProvider[IO]): Option[Rewards[IO, CurrencySnapshotStateProof, CurrencyIncrementalSnapshot, CurrencySnapshotEvent]] =
-    DorRewards.make[IO].some
+  override def rewards(implicit sp: SecurityProvider[IO]): Option[Rewards[IO, CurrencySnapshotStateProof, CurrencyIncrementalSnapshot, CurrencySnapshotEvent]] = {
+    val dailyBountyRewards = new DailyBountyRewards[IO]
+    val analyticsBountyRewards = new AnalyticsBountyRewards[IO]
+    val l0Url = getEnv[IO]("METAGRAPH_L0_NODE_URL")
+    val l1Url = getEnv[IO]("DATA_L1_NODE_URL")
```

Comment:
This doesn't change the effective functionality -- you're still executing the getEnv on every call. These should be higher up.

I see the confusion though -- we aren't defining `Main` like most apps do, so you need to put this in the `makeL0Service` so you have access to the results, then pass that to whoever needs it. E.g.
```
  private def makeL0Service: IO[BaseDataApplicationL0Service[IO]] = {
    for {
      l0Url <- getEnv[IO]("METAGRAPH_L0_NODE_URL")
      l1Url <- getEnv[IO]("DATA_L1_NODE_URL")
      validatorApi = new ValidatorNodesAPI[IO](l0Url, l1Url)
      //  then you can pass the api where needed
      calculatedStateService <- CalculatedStateService.make[IO]
      dataApplicationL0Service = makeBaseDataApplicationL0Service(calculatedStateService)
    } yield dataApplicationL0Service
  }
```

Reply:
@sdavidp the rewards function does not belong to this function, I mean it's not from `BaseDataApplicationL0Service`

Reply:
@IPadawans I see now. Ok, as discussed we're stuck having to do that on each invocation.